syntax = "proto3";

package commu;

service Commu{
  rpc send_index(threemess) returns (reply){}
}

//第一版的request
message index{
  repeated int32 _index = 1;
}

//第二版的request
message threemess{
  //前向dfv信息
  repeated int32 dfv_index = 1;
  repeated int32 value = 2;
  
  //node信息
  repeated int32 task_index = 3;

  //后向dfv信息，用于client(executor)返回信息给master
  repeated int32 dfv_index_backward = 4;

  int32 partition = 5;//表示哪一个分区

  int32 executor_number = 6;//表示哪一个client(executor),用于client知道自己是几号，在reply中返回给master
}

message reply{
  int32 ret = 1;

  //后向dfv信息
  repeated int32 dfv_index = 2;
  repeated int32 value = 3; 

  int32 partition = 4;//表示哪一个分区

  //表示哪一个client(executor);
  int32 executor_number = 5;
}

